src/honey_badger/epoch_state.rs

```diff
@@ -219,12 +222,22 @@ where
     /// Checks whether the subset has output, and if it does, sends out our decryption shares.
     fn process_subset(&mut self, cs_step: cs::Step<N>) -> Result<Step<C, N>> {
         let mut step = Step::default();
-        let mut cs_outputs = step.extend_with(cs_step, |cs_msg| {
+        let mut cs_outputs: VecDeque<_> = step.extend_with(cs_step, |cs_msg| {
             MessageContent::Subset(cs_msg).with_epoch(self.epoch)
         });
         if let Some(cs_output) = cs_outputs.pop_front() {
-            self.subset = SubsetState::Complete(cs_output.keys().cloned().collect());
-            step.extend(self.send_decryption_shares(cs_output)?);
+            match cs_output {
+                SubsetOutput::Contribution(k, v) => {
+                    let mut map: BTreeMap<N, Vec<u8>> = Default::default();
+                    self.keys.insert(k.clone());
+                    map.insert(k, v);
+                    step.extend(self.send_decryption_shares(map)?);
```

Comment:
Is this correct?  Should I be sending the decryption shares at this point, or is this premature?